mm2src/coins/eth.rs

```diff
@@ -3468,6 +3447,15 @@ pub async fn eth_coin_from_conf_and_request(
     let gas_station_policy: GasStationPricePolicy =
         json::from_value(req["gas_station_policy"].clone()).unwrap_or_default();
 
+    let key_lock = match &coin_type {
+        EthCoinType::Eth => String::from(ticker),
+        EthCoinType::Erc20 { ref platform, .. } => String::from(platform),
+    };
+
+    let mut map = NONCE_LOCK.lock().unwrap();
+
+    let nonce_lock = Arc::clone(map.entry(key_lock).or_insert_with(|| Arc::new(AsyncMutex::new(()))));
```

Comment:
Way too many brackets here :) Please create a helper
```rust
#[inline]
fn new_nonce_lock() -> Arc<AsyncMutex<()>> {
    Arc::new(AsyncMutex::new(()))
}
```
And refactor this specific line to
```rust
map.entry(key_lock).or_insert_with(new_nonce_lock).clone()
```

Reply:
done